src/native_mux_impl/command.rs

```diff
@@ -71,6 +71,10 @@ impl Command {
             stderr.as_raw_fd_or_null_fd()?,
         ];
 
+        for stdio in stdios {
+            set_blocking(stdio).map_err(Error::ChildIo)?;
+        }
```

Comment:
This is much better, thank you! One last request: can we add a comment here along the lines of:

> Note that once we do this, these file descriptors (and the `Fd`s we got them from above) should no longer be used in any async context, as they'd start blocking. We give away the descriptors in `stdios` when we pass them to `open_new_session` (which doesn't use them in an async context), and the `Fd`s are dropped when this function returns, meaning no owned file descriptors we set to be blocking here can be used in an async context in the future.

Reply:
Actually, writing the above made me realize that this isn't true for borrowed FDs, which include `StdioImpl::Fd` and, more importantly, `StdioImpl::Inherit`. This code, as written, will change the program's standard in and out to be blocking. Now, I _think_ that's okay for tokio given the not about a separate blocking thread [in their docs](https://docs.rs/tokio/latest/tokio/io/struct.Stdin.html), but it's probably unexpected for callers. And doubly so for borrowed FDs. I realize we may not have a choice, but I think this needs to be documented more explicitly!

Reply:
It's ok for `StdioImpl::Fd` since they owned the fd and `Stdio` is only used in `Command::spawn`.

For `StdioImpl::Inherit`, that is a problem I did not consider, maybe I can skip remolving non-blocking flags specifically for inherit mode.

Reply:
Done!

I've added the comment and also skip `set_blocking` for `StdioImpl::Inherit`.

Reply:
Hmm, is that true for `Fd::Borrowed` as well? It doesn't look like it is.

Also, why is it okay to skip this setting on the stdio fds? Won't the mux server still complain if those are non blocking too?

This also makes me skeptical of [this `impl`](https://github.com/openssh-rust/openssh/blob/master/src/stdio.rs#L49), which I think means any raw fd (including one that isn't owned) can be turned into one that *looks* owned (which is a problem).

Reply:
> Hmm, is that true for Fd::Borrowed as well? It doesn't look like it is.

I only skipped `set_blocking` for inherit.

> Also, why is it okay to skip this setting on the stdio fds? Won't the mux server still complain if those are non blocking too?

Well, that's indeed a problem, but I think it's bad to change fd 0, 1, 2 since they are considered as global state?

> This also makes me skeptical of [this impl](https://github.com/openssh-rust/openssh/blob/master/src/stdio.rs#L49), which I think means any raw fd (including one that isn't owned) can be turned into one that looks owned (which is a problem).

Hmmm, yeah.
Now looking at that it's also a problem.

Maybe we should only `set_blocking` for the `PipeRead`/`PipeWrite` we explicitly created?

Reply:
@jonhoo I've taken the route where we only call `set_blocking` fd we are 100% is owned, which are:
 - fds constructed from `OwnedFd`
 - fds constructed from other owned type, such as `PipeWrite`/`PipeRead`, `File`, `ChildStd*`.
 - `StdioImpl::Pipe`

For `StdioImpl::Inherit` and fds constructed from `FromRawFd::from_raw_fd`, `set_blocking` is not called and instead I added documentation for them.